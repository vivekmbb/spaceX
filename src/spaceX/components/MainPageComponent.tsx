import { Box, Grid, ResponsiveContext } from "grommet";
import React from "react";
import { RocketCard } from "./RocketCard";

interface IMainPageComponent {
  result: any;
}

export const MainPageComponent = ({ result }: IMainPageComponent) => {
  const size = React.useContext(ResponsiveContext);
  const count = size === "large" ? 4 : size === "medium" ? 2 : 1;
  const [newData, setnewData] = React.useState<any>();

  React.useEffect(() => {
    const constructedData = chunkArray(result, count);
    setnewData(constructedData);
  }, [size, result, count]);

  const chunkArray = (myArray: any[], chunk_size: number) => {
    let results = [];
    const newArray = [...myArray];
    while (newArray?.length) {
      results.push(newArray.splice(0, chunk_size));
    }
    return results;
  };

  const getGrids = () => {
    return newData?.map((object: any[], index: number) => (
      <Grid
        columns={{
          count,
          size: "auto"
        }}
        key={index}
        gap="10px"
        responsive={true}
        margin="xsmall"
        fill={true}
      >
        {getChildren(object)}
      </Grid>
    ));
  };

  const getChildren = (childrenData: any) => {
    return childrenData?.map((object: any, index: number) => <RocketCard key={index} data={object} />);
  };

  return newData ? <Box>{getGrids()}</Box> : null;
};
