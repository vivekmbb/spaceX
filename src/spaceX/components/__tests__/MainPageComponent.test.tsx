// tslint:disable: no-submodule-imports
import "@testing-library/jest-dom/extend-expect";
import { Box } from "grommet";
import React from "react";
import renderer from "react-test-renderer";
import { MainPageComponent } from "../MainPageComponent";

describe("Main Page Component", () => {
  const rocketinformation = [
    {
      flight_number: 6,
      mission_name: "Falcon 9 Test Flight",
      mission_id: ["EE86F74"],
      launch_year: "2010",
      launch_site: {
        site_name_long: "Cape Canaveral Air Force Station Space Launch Complex 40"
      },
      launch_success: true,
      links: {
        mission_patch_small: "https://images2.imgbox.com/5c/36/gbDKf6Y7_o.png"
      }
    }
  ];

  it("Main Page component", () => {
    const Wrapper = (props: any) => <Box pad="medium">{props.children}</Box>;
    const component = renderer
      .create(
        <Wrapper>
          <MainPageComponent result={rocketinformation} />
        </Wrapper>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
