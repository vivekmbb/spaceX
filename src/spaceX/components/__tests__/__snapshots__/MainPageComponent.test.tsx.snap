// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page Component Main Page component 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 chkvLT"
/>
`;
